libtelemetry/src/full/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -412,13 +413,34 @@ public void run() {
     });
   }
 
-  private static synchronized void setAccessToken(@NonNull Context context, @NonNull String accessToken) {
+  private static synchronized void setAccessToken(@NonNull final Context context,
+                                                  @NonNull String accessToken,
+                                                  @NonNull ExecutorService executorService) {
     if (TelemetryUtils.isEmpty(accessToken)) {
       return;
     }
     if (sAccessToken.getAndSet(accessToken).isEmpty()) {
```

Comment:
Add comment: // set the token and check the return value to see if it was previously nil, send pending error reports if we could not due to lack of token